full-service/src/service/account.rs

```diff
@@ -482,29 +513,81 @@ where
             first_block_index,
         );
 
-        let view_private_key_hex = hex::decode(view_private_key)?;
-        let view_private_key = RistrettoPrivate::try_from(view_private_key_hex.as_slice())?;
-
-        let spend_public_key_hex = hex::decode(spend_public_key)?;
-        let spend_public_key = RistrettoPublic::try_from(spend_public_key_hex.as_slice())?;
-
-        let import_block_index = self.ledger_db.num_blocks()? - 1;
-
         let mut pooled_conn = self.get_pooled_conn()?;
         let conn = pooled_conn.deref_mut();
+        let import_block_index = self.ledger_db.num_blocks()? - 1;
+
+        let view_account_key =
+            ViewAccountKey::new(*view_private_key.as_ref(), *spend_public_key.as_ref());
+
         exclusive_transaction(conn, |conn| {
             Ok(Account::import_view_only(
-                &view_private_key,
-                &spend_public_key,
+                &view_account_key,
                 name,
                 import_block_index,
                 first_block_index,
                 next_subaddress_index,
+                false,
                 conn,
             )?)
         })
     }
 
+    async fn import_view_only_account_from_hardware_wallet(
+        &self,
+        name: Option<String>,
+        first_block_index: Option<u64>,
+        fog_info: Option<FogInfo>,
+    ) -> Result<Account, AccountServiceError> {
+        let view_account = get_view_only_account_keys().await?;
+
+        let view_account_keys = ViewAccountKey::new(
+            *view_account.view_private_key().as_ref(),
+            *view_account.spend_public_key().as_ref(),
+        );
+
+        let mut pooled_conn = self.get_pooled_conn()?;
+        let conn = pooled_conn.deref_mut();
+        let import_block_index = self.ledger_db.num_blocks()? - 1;
+
+        match fog_info {
+            Some(fog_info) => {
+                let fog_authority_spki =
+                    general_purpose::STANDARD.decode(fog_info.authority_spki)?;
+                let default_subaddress_keys =
+                    get_view_only_subaddress_keys(DEFAULT_SUBADDRESS_INDEX).await?;
+
+                let default_public_address = get_public_fog_address(
+                    &default_subaddress_keys,
+                    fog_info.report_url,
+                    fog_info.report_id.unwrap_or_default(),
+                    &fog_authority_spki,
+                );
+                exclusive_transaction(conn, |conn: &mut diesel::SqliteConnection| {
```

Comment:
Optional/Nit: Do we need to specify the type here? Other references don't seem to.
```suggestion
                exclusive_transaction(conn, |conn| {
```